plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -238,10 +220,8 @@ func (e *Elasticsearch) gatherNodeStats(url string, acc telegraf.Accumulator) er
 			"cluster_name": nodeStats.ClusterName,
 		}
 
-		if e.ClusterStats {
-			// check for master
-			e.isMaster = (id == e.catMasterResponseTokens[0])
-		}
+		// check for master
+		e.localNodeIsMaster = (id == e.masterNodeId)
```

Comment:
I think the real problem here is that since we run the gather functions concurrently on all servers there is a race condition as to which server sets the value on the struct.  This is a current bug and should be resolved on a pull request separate from this one.